Sources/ArcGISToolkit/FloorFilter/SiteFacilityPromptViewController.swift

```diff
@@ -0,0 +1,272 @@
+//
+// Copyright 2021 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+import Foundation
+
+/// ViewController for the site and facility prompt
+internal class SiteFacilityPromptViewController: UIViewController {
+    var delegate: FloorFilterViewControllerDelegate?
+    var viewModel = FloorFilterViewModel()
+    
+    /// UI Elements and constraints
+    @IBOutlet weak var siteFacilitySearchBar: UISearchBar!
+    @IBOutlet weak var backBtn: UIImageView!
+    @IBOutlet weak var closeBtn: UIImageView!
+    @IBOutlet weak var promptTitle: UILabel!
+    @IBOutlet weak var promptSubtitle: UILabel!
+    @IBOutlet weak var promptTitleSubtitleStackView: UIStackView!
+    @IBOutlet weak var siteFacilityTableView: UITableView!
+    @IBOutlet weak var bottomConstraint: NSLayoutConstraint!
+    @IBOutlet weak var designableViewHeight: NSLayoutConstraint!
+    
+    private var originalYPositionForPanel: CGFloat = 0.0
+    
+    /// Show the facilities list directly if the map has no sites configured or if there is a previously selected facility
+    private var isShowingFacilities = false
+    private var isSearchActive = false
+    
+    /// Filtered facilities and sites list based on search query
+    private var filteredSearchFacilities: [AGSFloorFacility] = []
+    private var filteredSearchSites: [AGSFloorSite] = []
+    
+    private func getFilteredFacilities() -> [AGSFloorFacility] {
+        return isSearchActive ? filteredSearchFacilities : viewModel.facilities
+    }
+    
+    private func getFilteredSites() -> [AGSFloorSite] {
+        return isSearchActive ? filteredSearchSites : viewModel.sites
+    }
+    
+    override public func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        isShowingFacilities = (viewModel.sites.isEmpty || viewModel.selectedFacility != nil) ? true : false
+        initializeButtonsClickListeners()
+        updatePromptViewHeight()
+        updatePromptTitle()
```

Comment:
These are all called in `viewDidLoad()`. Do they all really need to be called again here?